modules/uncertainty/R/sensitivity.R

```
@@ -214,15 +215,57 @@ write.sa.configs <- function(defaults, quantile.samples, settings, model,
     ensemble.id <- settings$sensitivity.analysis$ensemble.id %||% rlang::hash(settings)
   }
   medianrun <- run.id
-
+  
   # create folders (cleaning up old ones if needed)
   if (clean) {
     unlink(file.path(settings$rundir, run.id))
     unlink(file.path(settings$modeloutdir, run.id))
   }
   dir.create(file.path(settings$rundir, run.id), recursive = TRUE)
   dir.create(file.path(settings$modeloutdir, run.id), recursive = TRUE)
-
+  
+  # I check to make sure the path under the met is a list.
+  # if it's specified what met needs to be used in 'met.id' under sensitivity
+  # analysis of pecan xml we used that otherwise, I use the first met.
+  if (is.list(settings$run$inputs$met$path)) {
+    # This checks for met.id tag in the settings under sensitivity analysis -
+    # if it's not there it creates it. Then it's gonna use what it created.
+    if (is.null(settings$sensitivity.analysis$met.id)) {
+      settings$sensitivity.analysis$met.id <- 1
```

👍 to the overall goal of picking one met (/other input) path to use across the whole SA. 

But calling this "met.id" is confusing to me -- looks like this is basically an index into the ensemble, whereas I would usually expect "met ID" to mean an identifier for which _source_ (e.g. CRUNCEP ERA5, etc) the met data came from. 

Could it instead be something like `settings$sensitivity.analysis$met_path` and contain the actual filepath to be used by the SA instead of an index that needs to be extracted from the ensemble? That'd be clearer to me, and would also allow a user to pass an arbitrary path to the SA even if it doesn't exist in the ensemble (not that I expect most people to want that, but it could come in handy sometimes)

---

...But wait! Doesn't `input_design` handle this for us just below?

---

good question, it duplicates with `input_design`( new approach )
* and also one of the reason is to be consistent with the ensemble pattern, so SA doing different (path) would be inconsistent
* and works when `input_design` is NOT provided (single-site or legacy workflows)
* The met.id resolution is indeed redundant when input_design is present. The met.id logic serves as a backward compatibility fallback for workflows that don't use `input_design`. However, when `input_design` exists it **supersedes** met.id and handles ALL inputs (met, IC, soil, etc.) . 
* Alternatively we could skip met.id resolution entirely when `input_design` exists


short-term fix (This PR):
keep current approach, add clarifying comment:
```
# select single met path for SA runs without input_design (backward compatibility)
# when input_design is provided (recommended), it supersedes this selection 
# and allows different met files per SA run 
if (is.list(settings$run$inputs$met$path)) { 
if (is.null(settings$sensitivity.analysis$met.id)) { 
settings$sensitivity.analysis$met.id <- 1 
} 
settings$run$inputs$met$path <- settings$run$inputs$met$path[[settings$sensitivity.analysis$met.id]] 
}
```
Long-term (Further iteration): 
Document `input_design` as recommended approach , consider deprecating met.id in favor of `input_design`.

---

Re met path: You've talked me out of the idea, especially since it would need to be site-specific -- so my proposal wouldn't need just one `settings$sensitivity.analysis$met_path`, it'd be at least a `settings$sensitivity.analysis$<site>$met_path` for every site. I do still suggest naming it something like `met.index` instead of `met.id`.

---

Re input design: I'll defer to @dlebauer but my sense is that since we've bought into making `input_design` mandatory for other parts of the workflow and it's specifically designed to be where we keep all the pieces of the sampling design together in one place, it may be sensible to make it mandatory here too rather than duplicate its functionality.

---

> I do still suggest naming it something like `met.index` instead of `met.id`

that's renaming to `met.index` would be nice clarity, but i am afraid of breaking backward compatibility for existing workflows may have `<met.id>` , and a direct rename would silently ignore those settings ?

---

I'm in favor of having a consistent `input_design` object. It is hard to see an advantage of different approaches if one generalize solution is available.

---

I am curious - where is met.id used? I can't find the string in either the PEcAn repository or in xml settings files used by CCMMF or Dongchen's workflows.

---

@dlebauer it's only used immediately below to set the met$path.

I agree with the consensus to drop the met.id solution proposed here and just use input_design. If one wants to only vary parameters and hold everything else fixed it's really easy to generate an input_design that's all 1's for everything other than parameters. The met.id "solution" isn't really a solution anyways as it only fixes met ensemble members and not the other ensembled inputs (phenology, soil physics, initial conditions, etc.)